source/volume_of_fluid/reconstruct.cc

```diff
@@ -0,0 +1,488 @@
+                    }
+                }
+            }
+
+            for (unsigned int i = 0; i < n_local; ++i)
+              {
+                if (neighbor_cells[i] == endc)
+                  {
+                    continue;
+                  }
+
+                fevalues.reinit(neighbor_cells[i]);
+
+                const std::vector<double> weights = fevalues.get_JxW_values();
+
+                double cell_vol = 0.0;
+                for (unsigned int j=0; j<weights.size(); ++j)
+                  {
+                    cell_vol+=weights[j];
+                  }
+
+                for (unsigned int nind = 0; nind < n_normals; ++nind)
+                  {
+                    const double normal_norm = normals[nind]*normals[nind];
+
+                    if (normal_norm > volume_fraction_threshold) // If candidate normal too small skip as set to max already
+                      {
+                        double dot = 0.0;
+                        for (unsigned int di = 0; di < dim; ++di)
+                          dot += normals[nind][di] * resc_cell_centers[i][di];
+                        const double n_volume_of_fluid = VolumeOfFluid::Utilities::compute_fluid_volume<dim> (max_degree, normals[nind], d_vals[nind]-dot,
+                                                         quadrature.get_points(), weights)/cell_vol;
+                        const double cell_err = local_volume_of_fluids (i) - n_volume_of_fluid;
+                        errs[nind] += cell_err * cell_err;
+                      }
+                  }
+              }
+
+            for (unsigned int nind = 0; nind < n_normals; ++nind)
+              {
+                if (errs[mn_ind] >= errs[nind])
+                  mn_ind = nind;
+                // std::cout << "\t" << normals[nind] << " d ";
+                // std::cout << d_vals[nind] << " e ";
+                // std::cout << errs[nind] << " " << mn_ind << std::endl;
+              }
+
+            normal = normals[mn_ind];
+            d = d_vals[mn_ind];
+          }
+
+        // double n2 = sqrt(normal*normal);
+        // if (n2 > volume_fraction_threshold)
+        //   {
+        //     normal = (normal / n2);
+        //     d = d / n2;
+        //   }
+        // else
+        //   {
+        //     normal[0] = 0.0;
+        //     normal[1] = 0.0;
+        //   }
+
+        for (unsigned int i=0; i<dim; ++i)
+          initial_solution (local_dof_indicies[system_fe
+                                               .component_to_system_index(volume_of_fluidN_c_index+i, 0)]) = normal[i];
+
+        initial_solution (local_dof_indicies[system_fe
+                                             .component_to_system_index(volume_of_fluidN_c_index+dim, 0)]) = d;
+
+        for (unsigned int i=0; i<n_volume_of_fluidLS_dofs; ++i)
+          {
+            // Recenter unit cell on origin
+            Tensor<1, dim, double> uSupp = volume_of_fluidLS_var.fe->unit_support_point(i)-uReCen;
```

Comment:
Rename uSupp to something for which I can understand what it is. `recentered_support_point`?